AmericanMedicalAssociation/AMA-style-guide
# Ewl 3574 EC Marketing Page template + Image Landscape card rows

## Ticket(s)

**Jira Ticket**

- [EWL-3574: EC | Create Education Center Marketing Page Template](https://issues.ama-assn.org/browse/EWL-3574)
- [EWL-3622: EC | Design Landscape Image Card & Row - style guide](https://issues.ama-assn.org/browse/EWL-3622)


## Description

Creates a template for the EC Marketing page, as well as the card rows for landscape cards.


## To Test
Observe that the following patterns display properly at all breakpoints:
- Molecules > Layouts > Three Up layout
- Molecules > Layouts > Four Up layout
- Molecules > Layouts > Five Up Left layout (when you merge this, please also delete the branch for closed PR https://github.com/AmericanMedicalAssociation/AMA-style-guide/pull/137)
- Molecules > Components > Image Landscape card
- Organisms > Sections > EC Card Row (Three-up)
- Organisms > Sections > EC Card Row (Four-up)
- Organisms > Sections > EC Card Row (Five-up Left)
- Templates > EC Home > EC Home


## Relevant Screenshots/GIFs
![screen shot 2017-06-19 at 10 29 59 am](https://user-images.githubusercontent.com/12160398/27292873-51697012-54da-11e7-8c93-7eb2c9b816d3.png)
(Templates > EC Home > EC Home)

![screen shot 2017-06-19 at 10 30 13 am](https://user-images.githubusercontent.com/12160398/27292875-53a571fa-54da-11e7-9b8d-5e10d1a14303.png)
(Organisms > Sections > EC Card Row (Five-up Left)



## Remaining Tasks
(I'll either address these today, or they can be broken out into separate tickets. Mostly noting them here to point out that I'm aware that they need to be fixed, but the work seemed either beyond the scope of this ticket or tweaks that could easily be made after the PR was submitted for review)
- [x] Gutters for the tablet (3-up) breakpoint of the Five Up Left layout pattern are uneven. (They're better now, but still uneven.)
- [ ] There's a right gap on the right side of the three-up layout, which already existed. 
- [ ] There are magic numbers in the height calc() declarations, as well as other places in the CSS. Either variable-ify them, or explain them in comments.
- [x] When viewing the EC Home template pattern, the vertical gutters are weird between the various layouts. 
- [ ] The three-up layout has negative margin/padding in the style of a classic grid. IMO this is confusing, and also it throws off the `<a>` hover style in the four-up and five-up layouts. Ideally, it would be nice to set up the Three Up layout so that it doesn't rely on negative margins/padding.
- [ ] Image Landscape Card pattern needs a height when viewing the pattern on its own.



## Additional Notes

~Dependent on PR #137.~ (Originally this was the case, but I'm afraid #137 isn't relevant anymore unless we decide we like that version of the Five Up Left layout pattern better)
